src/handlers/shared/start.ts

```diff
@@ -147,6 +147,34 @@ export async function start(
         });
       });
     }
+
+    if (priceLabel && role !== "admin") {
+      const { usdPriceMax } = taskAccessControl;
+      const min = Math.min(...Object.values(usdPriceMax));
+      const userAllowedMaxPrice = !role ? min : usdPriceMax[role as keyof typeof usdPriceMax];
+
+      const priceRegex = /Price:\s*([\d.]+)/;
+      const match = priceLabel.name.match(priceRegex);
+      if (!match) {
+        throw logger.error("Price label is not in the correct format", { priceLabel: priceLabel.name });
+      }
+      const value = match[1];
+      if (isNaN(parseFloat(value))) {
+        throw logger.error("Price label is not in the correct format", { priceLabel: priceLabel.name });
+      }
+      const price = parseFloat(value);
+      if (price > userAllowedMaxPrice) {
+        throw logger.error(
```

Comment:
```suggestion
        throw logger.warn(
```
only use error for errors that are not expected